AutoNode/validator.py

```diff
@@ -0,0 +1,170 @@
+import sys
+import time
+import json
+
+from pyhmy import cli
+from pyhmy import (
+    Typgpy,
+)
+
+from .common import (
+    bls_key_dir,
+    node_config,
+    validator_config,
+    saved_wallet_pass_path,
+    harmony_dir,
+)
+from .blockchain import (
+    get_latest_header,
+    get_latest_headers,
+    get_staking_epoch,
+    get_current_epoch,
+    get_all_validator_addresses,
+    get_validator_information
+)
+from .node import (
+    wait_for_node_response,
+    assert_no_bad_blocks
+)
+from .util import (
+    check_min_bal_on_s0,
+    input_with_print
+)
+
+log_path = f"{harmony_dir}/autonode_validator.log"
+
+
+def _add_bls_key_to_validator():
+    wait_for_node_response(node_config['endpoint'], verbose=True)
+    chain_val_info = get_validator_information(validator_config['validator-addr'], node_config['endpoint'])
+    bls_keys = set(x.replace('0x', '') for x in chain_val_info["validator"]["bls-public-keys"])
+    for k in (x.replace('0x', '') for x in node_config["public-bls-keys"]):
+        if k not in bls_keys:  # Add imported BLS key to existing validator if needed
+            print(f"{Typgpy.OKBLUE}Adding bls key: {Typgpy.OKGREEN}{k}{Typgpy.OKBLUE} "
+                  f"to validator: {Typgpy.OKGREEN}{validator_config['validator-addr']}{Typgpy.ENDC}")
+            try:
```

Comment:
in general - i think some of your for/try might be better as try/for

Reply:
will keep an eye out when refactoring